extension/src/experiments/data/index.ts

```diff
@@ -56,21 +56,16 @@ export class ExperimentsData extends BaseData<ExperimentsOutput> {
   private async updateExpShow() {
     await this.updateBranches()
     const branches = this.experiments.getBranchesToShow()
-    let gitLog = ''
-    const rowOrder: { branch: string; sha: string }[] = []
     const availableNbCommits: { [branch: string]: number } = {}
-    const args: Args = []
 
+    const promises = []
     for (const branch of branches) {
-      gitLog = await this.collectGitLogAndOrder(
-        gitLog,
-        branch,
-        rowOrder,
-        availableNbCommits,
-        args
-      )
+      promises.push(this.collectGitLogByBranch(branch, availableNbCommits))
     }
 
+    const branchLogs = await Promise.all(promises)
+    const { args, gitLog, rowOrder } = this.collectGitLogAndOrder(branchLogs)
```

Comment:
[F] The order that we collect the information from the promises is important so we need to wait for them all to resolve before we can proceed. This should still give us a performance boost for a large number of branches.